src/dune_rules/dune_file.ml

```diff
@@ -1775,9 +1776,19 @@ module Copy_files = struct
     let+ alias = field_o "alias" (check >>> Alias.Name.decode)
     and+ mode =
       field "mode" ~default:Rule.Mode.Standard (check >>> Rule.Mode.decode)
+    and+ enabled_if =
+      let since = (2, 8) in
+      let allowed_vars = Enabled_if.common_vars ~since in
```

Comment:
Why isn't `allowed_vars = Any`?

Reply:
Good question... I copied what was done for some of the other stanzas without giving it much thought. I will change this to `Any`.

By the way, why is the set of allowed variables restricted for some of the stanzas (executables, libraries, install) ?

Reply:
This is fixed now.

Reply:
I think it was for some compatibility reasons. We accidentally introduced variables in one field without allowing them in a similar field (@voodoos might want to confirm, I don't remember this so much anymore). For new fields, I think the better default it to allow everything.

Reply:
I myself followed what was already done for libraries when adding the field for executables.
The idea was (if I remember well) to prevent users from relying on some dune inner-workings like the _build folder organization. 

However I think that a "blacklist" of variables would make more sense than the actual "whitelist".

@jeremiedimino should know more about the historical reasons of the variable restriction in the libraries `enabled_if` field.

Reply:
Let's take an example: `%{library-avaiable:<lib>}`. To answer this question, we must have the library database, to have the library database, we must know the set of defined libraries.

So if you have `%{library-available:<lib>}` in the `enabled_if` field of a library, then you get a computation cycle. To avoid this kind of problems, we leaned on the conservative side by using a white list.